py3status/modules/clementine.py

```diff
@@ -28,54 +30,55 @@ class Py3status:
     cache_timeout = 5
     format = u'♫ {current}'
 
-    def _getMetadatas(self):
+    def clementine(self):
         """
-        Get the current song metadatas (artist - title)
+        Get current song metadata: "artist - title"
         """
-        track_id = check_output(CMD + '.GetCurrentTrack', shell=True)
-        metadatas = check_output(
-            CMD + '.GetMetadata {}'.format(track_id.decode()), shell=True
-        )
-        lines = metadatas.decode('utf-8').split('\n')
-        lines = filter(None, lines)
-
-        now_playing = ''
-
-        if lines:
-            artist = ''
+        if not self.py3.check_commands(['clementine']):
+            return {
+                'cached_until': self.py3.CACHE_FOREVER,
+                'color': self.py3.COLOR_BAD,
+                'full_text': STRING_UNAVAILABLE
+            }
+
+        artist = lines = now_playing = title = ''
+        internet_radio = False
+
+        try:
+            track_id = self.py3.command_output(CMD + '.GetCurrentTrack')
+            metadata = self.py3.command_output(CMD + '.GetMetadata {}'.format(track_id))
+            lines = filter(None, metadata.splitlines())
+        except:
+            return {
+                'cached_until': self.py3.time_in(self.cache_timeout),
+                'color': self.py3.COLOR_BAD,
+                'full_text': STRING_ERROR
+            }
+
+        for item in lines:
+            if item.find('artist:') != -1:
```

Comment:
I don't really like all this `if item.find('artist:') != -1:` we could just do `if 'artist' in item:` to me this is more readable - you can keep the `find()` if you want.